src/deepsparse/transformers/pipelines/pipeline.py

```diff
@@ -104,30 +120,51 @@ def sequence_length(self) -> Union[int, List[int]]:
 
     def setup_onnx_file_path(self) -> str:
         """
-        Parses ONNX, tokenizer, and config file paths from the given `model_path`.
-        Supports sparsezoo stubs
+        Parses ONNX model from the `model_path` provided.
+        For tokenizers and model configs, supports paths, dictionaries,
+        or transformers.PretrainedConfig/transformes.PreTrainedTokenizerBase types. Also
+        supports the default None, in which case the config and tokenizer are read from
+        the provided `model_path`.
 
-        :param delay_overwriting_inputs: if True, do not overwrite the ONNX model
-            inputs to the given sequence length. Default is False
+        Supports sparsezoo stubs
 
         :return: file path to the processed ONNX file for the engine to compile
         """
-        onnx_path, config_path, tokenizer_path = get_onnx_path_and_configs(
-            self.model_path, require_configs=True
-        )
+        onnx_path = get_onnx_path(self.model_path)
+
+        if not self.config or not self.tokenizer:
+            config_found, tokenizer_found = get_hugging_face_configs(self.model_path)
+            if config_found:
+                self.config = config_found
+            if tokenizer_found:
+                self.tokenizer = tokenizer_found
+
+        if isinstance(self.config, dict):
+            local_config_path = os.path.join(self.model_path, "config.json")
+            with open(local_config_path, "w") as f:
```

Comment:
do we really need to save an intermediate file to disk?

Reply:
From the hugging face docs, it seems like the `AutoConfig.from_pretrained` requires a file path of some sort: https://huggingface.co/docs/transformers/v4.32.1/en/model_doc/auto#transformers.AutoConfig.from_pretrained.pretrained_model_name_or_path but definitely open to not saving if there's a way around it.